ec/resources/dnsroute/secondary_zone.go

```diff
@@ -275,62 +340,93 @@ func ResourceSecondaryZoneGroupUpdate(ctx context.Context, d *schema.ResourceDat
 		s := server[0].(map[string]interface{})
 		sID := s["master_server_id"].(int)
 
-		ms := api.MasterServerRequest{
+		ms := routedns.MasterServer{
 			ID: sID,
 		}
 
 		t := tsig[0].(map[string]interface{})
 		tID := t["tsig_id"].(int)
-		ts := api.TsigRequest{
+		ts := routedns.TSIGGetOK{
 			ID: tID,
 		}
 
-		mstItem := api.MasterServerTsigRequest{
+		mstItem := routedns.MasterServerTSIG{
 			MasterServer: ms,
-			Tsig:         ts,
+			TSIG:         ts,
 		}
 		mstArr = append(mstArr, mstItem)
 	}
 
-	zcr := api.DnsZoneCompositionRequest{
+	zcr := routedns.ZoneCompositionResponse{
 		MasterGroupID:     masterGroupID,
 		MasterServerTsigs: mstArr,
 		Zones:             zoneArr,
 	}
 
-	secondaryZoneGroupRequest := &api.DnsRouteSecondaryZoneGroupRequest{
-		ID:              secondaryZoneGroupID,
-		Name:            name,
-		ZoneComposition: zcr,
-		CustomerID:      -1,
+	// Retrieve Existing Secondary Zone Group Object
+	getParams := routedns.NewGetSecondaryZoneGroupParams()
+	getParams.AccountNumber = accountNumber
+	getParams.ID = secondaryZoneGroupID
+	groupObj, err := routeDNSService.GetSecondaryZoneGroup(*getParams)
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
-	dnsRouteClient := api.NewDNSRouteAPIClient(*config)
-	err = dnsRouteClient.UpdateSecondaryZoneGroup(secondaryZoneGroupRequest)
+	// Update Secondary Zone Group Object
+	groupObj.Name = name
+	groupObj.ZoneComposition = zcr
+
+	// Call Update Secondary Zone Group API
+	updateParams := routedns.NewUpdateSecondaryZoneGroupParams()
+	updateParams.AccountNumber = accountNumber
+	updateParams.SecondaryZoneGroup = *groupObj
+
+	err = routeDNSService.UpdateSecondaryZoneGroup(*updateParams)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	return ResourceSecondaryZoneGroupRead(ctx, d, m)
 }
 
-func ResourceSecondaryZoneGroupDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	var diags diag.Diagnostics
+func ResourceSecondaryZoneGroupDelete(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Initialize Route DNS Service
+	secondaryZoneGroupID, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
 	accountNumber := d.Get("account_number").(string)
 	config := m.(**api.ClientConfig)
-	(*config).AccountNumber = accountNumber
-
-	dnsRouteAPIClient := api.NewDNSRouteAPIClient(*config)
+	routeDNSService, err := buildRouteDNSService(**config)
+	if err != nil {
+		return diag.FromErr(err)
+	}
 
-	szgID, _ := strconv.Atoi(d.Id())
+	// Call Get Secondary Zone Group API
+	getParams := routedns.NewGetSecondaryZoneGroupParams()
```

Comment:
Similar to MSG - do we need to ensure the Zone exists before deleting?

Reply:
This is an existing pattern in the provider but I will being a topic to discuss with the team with some pros and cons of this approach. Thanks for mentioning it. Would prefer to keep this for consistency for now.